std/xml.d

```diff
@@ -2045,8 +2045,8 @@ class ElementParser
 
                     immutable(char)* p = startTag.tagString.ptr
                         + startTag.tagString.length;
-                    immutable(char)* q = tag_.tagString.ptr;
-                    text = decode(p[0..(q-p)], DecodeMode.LOOSE);
+                    immutable end = &tag_.tagString[0] - (&startTag.tagString[0] + startTag.tagString.length);
+                    text = decode(p[0 .. end], DecodeMode.LOOSE);
```

Comment:
Oh my. I didn't notice that p was a pointer still.

Yeah, I feel this change is half way, but gives us no better results. In fact it gives us *less* performance, as we are now doing bounds checking on the first element. And we still can't call this function `@safe`.

I'm now looking to see where `s` is defined. My god, it's a *member* variable string *pointer*. Who calls their members `s`?

@JackStouffer I know you are trying to move this into a `@safe` direction, but this code may be unfixable.